app/models/spotlight/role.rb

```diff
@@ -8,6 +8,15 @@ class Role < ActiveRecord::Base
     validates :role, inclusion: { in: ROLES }
     validate :user_must_be_unique, if: :user
 
+    after_create do
+      # Deliver an invitation to the user if their account
+      # was created by an invite and have not accepted it yet.
+      # This is being done after create because the user and role
+      # both need to be persisted given that the role information
+      # is used in the content of the invitation email.
+      user.deliver_invitation if user.created_by_invite? && user.invitation_accepted_at.blank?
```

Comment:
This doesn't really feel like this right place for this to be to me, but this appears to be in the spirit of what the refactoring of this code was intended to do.

Reply:
Yeah, doing this in the model isn't awesome. Maybe it's worth looking into the `RolesController` and figuring out if we have enough information there to send out invites after the models are created?